qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -226,6 +240,26 @@ def result(self, timeout: Optional[float] = None, wait: float = 5) -> Result:
 
         return self._result
 
+    def _retrieve_partial_result(self) -> Optional[Result]:
+        """Attempt to retrieve partial results for a job.
+
+        Returns:
+            The partial results for a job, or ``None`` if schema validation fails.
+        """
+        if not self._result and not self._partial_result:  # type: ignore[misc]
+            # No results, nor partial results, have been attained for this job yet.
+            with api_to_job_error():
+                # Set partial results, in case validation fails they are cached.
+                self._partial_result = self._api.job_result(self.job_id(), self._use_object_storage)
```

Comment:
I think `result_response` is a better name for this, since the data returned by `job_result()` may or may not contain partial results. It's also a bit confusing about the differences between `_partial_result` and `_result` (since `_result` could also be partial).